src/types/boolean8.rs

```diff
@@ -0,0 +1,42 @@
+use std::cmp::Ordering;
+use std::convert::TryInto;
+
+use crate::compressor::Compressor;
+use crate::decompressor::Decompressor;
+use crate::types::{DataType, NumberLike};
+
+impl NumberLike for bool {
+  fn num_eq(&self, other: &Self) -> bool {    
+    self.eq(other)
+  }
+
+  fn num_cmp(&self, other: &Self) -> Ordering {
+    self.cmp(other)
+  }
+}
+
+pub struct B8DataType {}
+
+impl DataType<bool> for B8DataType {
+  const HEADER_BYTE: u8 = 4;
```

Comment:
Oh yeah, this byte just needs to be something unique. `u32` currently uses 4, so we should set this to something untaken, like 7. This just makes sure that when you read in `q_compressed` data you won't get mixed up about which data type it is